setup-env

```diff
@@ -9,61 +9,121 @@ USAGE=$(
 Configure a development environment for this repository.
 
 It does the following:
+  - Allows the user to specify the Python version to use for the virtual environment.
+  - Allows the user to specify a name for the virtual environment.
   - Verifies pyenv and pyenv-virtualenv are installed.
-  - Creates a Python virtual environment.
+  - Creates the Python virtual environment.
   - Configures the activation of the virtual enviroment for the repo directory.
   - Installs the requirements needed for development.
   - Installs git pre-commit hooks.
   - Configures git upstream remote "lineage" repositories.
 
 Usage:
-  setup-env [options] [virt_env_name]
+  setup-env (-n | --name) [virtual_env_name] (-v | --version) [python_version]
   setup-env (-h | --help)
 
 Options:
-  -f --force          Delete virtual enviroment if it already exists.
-  -h --help           Show this message.
-  -i --install-hooks  Install hook environments for all environments in the
-                      pre-commit config file.
+  -f | --force             Delete virtual enviroment if it already exists.
+  -h | --help              Show this message.
+  -i | --install-hooks     Install hook environments for all environments in the
+                           pre-commit config file.
+  -l | --list-versions     List available Python versions and select interactively.
+  -n | --name              Specify the name of the virtual environment.
+  -v | --version           Specify the Python version for the virtual environment.
 
 END_OF_LINE
 )
 
+# Display installed python versions
+python_versions() {
+  pyenv versions --bare --skip-aliases --skip-envs
+}
+
 # Flag to force deletion and creation of virtual environment
 FORCE=0
 
-# Positional parameters
-PARAMS=""
+# Initialize the other flags
+INSTALL_HOOKS=0
+LIST_VERSIONS=0
+PYTHON_VERSION=""
+VENV_NAME=""
+
+# Define long options
+LONGOPTS="force,help,install-hooks,list-versions,name:,version:"
+
+# Define short options for getopt
+SHORTOPTS="fhiln:v:"
+
+# Check if GNU getopt is available
+if [[ $(getopt --version 2> /dev/null) != *"getopt from util-linux"* ]]; then
+  cat << 'END_OF_LINE'
+
+  Gnu-getopt is not detected and is a dependency to run this script.
+  On the Mac, we recommend installing brew, https://brew.sh/.  Then installation
+  is as simple as `brew install gnu-getopt` and adding this to your
+  profile:
+
+  export PATH="$(brew --prefix)/opt/gnu-getopt/bin:$PATH"
```

Comment:
@mcdonnnj were we going to leave this as is based on the previous conversation above?

Reply:
Like my suggestion in https://github.com/cisagov/skeleton-generic/pull/166#discussion_r1497936838 I would like to make reference of the fact that this is necessary because the `gnu-getopt` formula is keg-only and so must be added to the PATH manually.

Reply:
Pardon my ignorance, but what does keg-only mean in this context?

Reply:
@jsf9k "When Homebrew installs a package as "keg-only," it means that the package is installed in its own directory (a "keg" in Homebrew terminology) but is not symlinked into /usr/local, meaning the executables won't be available in the system's PATH by default. This approach is typically used for packages that could conflict with system binaries or libraries if they were symlinked globally." -ChatGPT

Reply:
@mcdonnnj See here: https://github.com/cisagov/skeleton-generic/pull/166/commits/88724e70167a5864495eccff5532f4c59fbefc1c